src/rgw/rgw_rest_s3.cc

```diff
@@ -661,45 +661,71 @@ void RGWGetUsage_ObjStore_S3::send_response()
   rgw_flush_formatter_and_reset(s, s->formatter);
 }
 
-int RGWListBucket_ObjStore_S3::get_params()
+
+int RGWListBucket_ObjStore_S3::get_common_params()
+
 {
-  list_versions = s->info.args.exists("versions");
-  prefix = s->info.args.get("prefix");
-  if (!list_versions) {
-    marker = s->info.args.get("marker");
-  } else {
-    marker.name = s->info.args.get("key-marker");
-    marker.instance = s->info.args.get("version-id-marker");
-  }
+list_versions = s->info.args.exists("versions");
+prefix = s->info.args.get("prefix");
+list_versions = s->info.args.exists("versions");
```

Comment:
These two statements have been repeated: list_versions = .. and prefix = ..

Reply:
Removed